src/main/java/org/openrewrite/java/spring/swagger/TransformApiInfo.java

```diff
@@ -0,0 +1,155 @@
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.function.BiFunction;
+import java.util.function.Function;
+
+public class TransformApiInfo extends Recipe {
+
+    @Override
+    public String getDisplayName() {
+        return "Transform SpringFox `ApiInfo` to Swagger v3 `Info`";
+    }
+
+    @Override
+    public String getDescription() {
+        return "Transforms SpringFox `ApiInfoBuilder` to Swagger v3 `Info` fluent API pattern.";
+    }
+
+    @Override
+    public TreeVisitor<?, ExecutionContext> getVisitor() {
+        return Preconditions.check(
+                new UsesType<>("springfox.documentation.builders.ApiInfoBuilder", true),
+                new TransformApiInfoVisitor()
+        );
+    }
+
+    private static class TransformApiInfoVisitor extends JavaIsoVisitor<ExecutionContext> {
+        private final MethodMatcher apiInfoBuilderConstructor = new MethodMatcher("springfox.documentation.builders.ApiInfoBuilder <constructor>()");
+        private final MethodMatcher termsOfServiceUrl = new MethodMatcher("springfox.documentation.builders.ApiInfoBuilder termsOfServiceUrl(String)");
+        private final MethodMatcher licenseUrl = new MethodMatcher("springfox.documentation.builders.ApiInfoBuilder licenseUrl(String)");
+        private final MethodMatcher license = new MethodMatcher("springfox.documentation.builders.ApiInfoBuilder license(String)");
+        private final MethodMatcher contact = new MethodMatcher("springfox.documentation.builders.ApiInfoBuilder contact(springfox.documentation.service.Contact)");
+        private final MethodMatcher contactConstructor = new MethodMatcher("springfox.documentation.service.Contact <constructor>(String, String, String)");
+        private final MethodMatcher build = new MethodMatcher("springfox.documentation.builders.ApiInfoBuilder build()");
+
+        // accumulator for AppInfoBuilder methods
+        private static class Accumulator {
+            private final Map<String, Function<J.MethodInvocation, String>> transformations = new HashMap<>();
+            private final Map<String, J.MethodInvocation> methodInvocations = new HashMap<>();
+            private final BiFunction<J.MethodInvocation, J.MethodInvocation, String> licenseTransformer =
+                    (name, url) -> {
+                        if (name == null && url == null) {
+                            return "";
+                        }
+                        StringBuilder sb = new StringBuilder(".license(new License()");
+                        if (name != null) {
+                            sb.append(".name(\"").append(name.getArguments().get(0)).append("\")");
+                        }
+                        if (url != null) {
+                            sb.append(".url(\"").append(url.getArguments().get(0)).append("\")");
+                        }
+                        sb.append(')');
+                        return sb.toString();
+                    };
```

Comment:
Not much point in a final lambda in this context, just declare a method.
```java
private final String licenseTransformer(J.MethodInvocation name, J.MethodInvocation url) { } 
```
That can be used in all the same ways a lambda can.